src/pages/workspaces/workspace/analysis/utils/cost-utils.ts

```diff
@@ -190,77 +217,85 @@ export const getAzureDiskCostEstimate = ({ region, persistentDiskSize }) => {
+  return price
 }
-export const getPersistentDiskCostHourly = ({ size, status, diskType, cloudContext = {} }, computeRegion) => {
-  const price = Utils.cond(
-    // @ts-expect-error
-    [isAzureContext(cloudContext), () => getAzureDiskCostEstimate({ persistentDiskSize: size, region: computeRegion }) / numberOfHoursPerMonth],
+export const getPersistentDiskCostHourly = (disk:DecoratedPersistentDisk|{ size:number; status:LeoDiskStatus; diskType:PdType; cloudContext?:CloudContext }, computeRegion:string):number => {
+  if (!disk || !computeRegion) return 0
+  const { size, status, diskType, cloudContext } = disk
+  let price = 0.0
+  price = _.includes(status, [diskStatuses.deleting.leoLabel, diskStatuses.failed.leoLabel]) ? 0.0 : Utils.cond(
+    [cloudContext && isAzureContext(cloudContext), () => getAzureDiskCostEstimate(computeRegion, size) / numberOfHoursPerMonth],
     [Utils.DEFAULT, () => size * getPersistentDiskPriceForRegionHourly(computeRegion, diskType)],
   )
-  return _.includes(status, ['Deleting', 'Failed']) ? 0.0 : price
+  return price
 }
 
-export const getRuntimeCost = ({ runtimeConfig, status, cloudContext = {} }) => {
-  // @ts-expect-error
-  if (isAzureContext(cloudContext)) {
+export const getRuntimeCost = (runtime:Runtime):number => {
+  if (!runtime) return 0
+  const { runtimeConfig, status } = runtime
+  if (isAzureConfig(runtimeConfig)) {
     return Utils.switchCase(status,
-      ['Stopped', () => 0.0],
-      ['Error', () => 0.0],
+      [runtimeStatuses.stopped.leoLabel, () => 0.0],
+      [runtimeStatuses.error.leoLabel, () => 0.0],
       [Utils.DEFAULT, () => getAzureComputeCostEstimate(runtimeConfig)]
     )
+  } else if (isGceRuntimeConfig(runtimeConfig) || isDataprocConfig(runtimeConfig)) {
+    return Utils.switchCase(status,
+      [
+        runtimeStatuses.stopped.leoLabel,
+        () => runtimeConfigBaseCost(runtimeConfig)
+      ],
+      [runtimeStatuses.error.leoLabel, () => 0.0],
+      [Utils.DEFAULT, () => runtimeConfigCost(runtimeConfig)]
+    )
+  } else {
+    throw new Error(`Unknown runtime config type ${runtimeConfig}`)
   }
-  return Utils.switchCase(status,
-    [
-      'Stopped',
-      () => runtimeConfigBaseCost(runtimeConfig)
-    ],
-    ['Error', () => 0.0],
-    [Utils.DEFAULT, () => runtimeConfigCost(runtimeConfig)]
-  )
 }
 
-export const getCostForDisk = (app, appDataDisks, computeRegion, currentRuntimeTool, persistentDisks, runtimes, toolLabel) => {
-  let diskCost = ''
-  const curPd = persistentDisks && persistentDisks.length && getCurrentPersistentDisk(runtimes, persistentDisks)
+export const getCostForDisk = (app:App, appDataDisks:PersistentDisk[], computeRegion:string, currentRuntimeTool:RuntimeToolLabel, persistentDisks:PersistentDisk[], runtimes:Runtime[], toolLabel:ToolLabel):number => {
+  let diskCost = 0
+  const rawPd = persistentDisks && persistentDisks.length && getCurrentPersistentDisk(runtimes, persistentDisks)
```

Comment:
(i don't know TS) is the ` pds && pds.length` check needed if the param has type PD[]? won't it only accept an array obj?

Reply:
you're right there, what it should be is pds && pds.length > 0